ci-operator/step-registry/cucushift/upgrade/paused/intermediate/to-latest/cucushift-upgrade-paused-intermediate-to-latest-commands.sh

```diff
@@ -0,0 +1,264 @@
+    fi        
+    
+    echo "Require admin ack"
+    local wait_time_loop_var=0 ack_data 
+    ack_data="$(echo ${out} | awk '{print $2}' | cut -f2 -d\")" && echo "Admin ack patch data is: ${ack_data}"
+    oc -n openshift-config patch configmap admin-acks --patch '{"data":{"'"${ack_data}"'": "true"}}' --type=merge
+    
+    echo "Admin-acks patch gets started"
+            
+    echo -e "sleep 5 min wait admin-acks patch to be valid...\n"
+    while (( wait_time_loop_var < 5 )); do
+        sleep 1m
+        echo -e "wait_time_passed=${wait_time_loop_var} min.\n"
+        if ! oc adm upgrade | grep "AdminAckRequired"; then
+            echo -e "Admin-acks patch PASSED\n"
+            return 0              
+        else
+            echo -e "Admin-acks patch still in processing, waiting...\n"
+        fi
+        (( wait_time_loop_var += 1 ))
+    done
+    if (( wait_time_loop_var >= 5 )); then
+        echo >&2 "Timed out waiting for admin-acks completing, exiting" && return 1
+    fi
+}
+
+function pause_worker_mcp() {
+    echo "Pausing worker pool..."
+    oc patch --type=merge --patch='{"spec":{"paused":true}}' machineconfigpool/worker
+    ret=$(oc get mcp worker -ojson | jq -r '.spec.paused')
+    if [[ "${ret}" != "true" ]]; then
+        echo >&2 "Worker pool failed to pause, exiting" && return 1
+    fi
+}
+
+function switch_channel() {
+    echo "Switch upgrade channel to eus-${TARGET_Y_VERSION}..."
+    oc adm upgrade channel eus-${TARGET_Y_VERSION}
+    ret=$(oc get clusterversion/version -ojson | jq -r '.spec.channel')
+    if [[ "${ret}" != "eus-${TARGET_Y_VERSION}" ]]; then
+        echo >&2 "Failed to switch channel, exiting" && return 1
+    fi
+}
+
+function upgrade() {
+    echo "Checking if there are available updates"
+    ret=$(oc get clusterversion/version -ojson | jq .status.availableUpdates)
+    if [[ "${ret}" == "null" ]]; then
+        echo >&2 "No available updates, exiting" && return 1
+    fi
+    cmd="oc adm upgrade --to-latest"
+    echo "Upgrading cluster: ${cmd}"
+    if ! eval "${cmd}"; then
+        echo >&2 "Failed to request upgrade, exiting" && return 1
+    fi
+}
+
+# Monitor the upgrade status
+function check_upgrade_status() {
+    local wait_upgrade="${TIMEOUT}" out avail progress
+    while (( wait_upgrade > 0 )); do
+        echo "oc get clusterversion" && oc get clusterversion
+        out="$(oc get clusterversion --no-headers)"
+        avail="$(echo "${out}" | awk '{print $3}')"
+        progress="$(echo "${out}" | awk '{print $4}')"
+        if [[ ${avail} == "True" && ${progress} == "False" && ${out} == *"Cluster version is ${TARGET_VERSION}" ]]; then
+            echo -e "Upgrade succeed\n\n"
+            return 0
+        else
+            sleep 5m
+            (( wait_upgrade -= 5 ))
+        fi        
+    done
+    if (( wait_upgrade <= 0 )); then
+        echo >&2 "Upgrade timeout, exiting" && return 1
+    fi
+}
+
+function run_command_oc() {
```

Comment:
No place to reference this function?

Reply:
Thanks! Updated.

Reply:
How about rename the function name to `run_command_oc_retries`

Reply:
Thanks for the suggestion! Updated.